File: P3.m
try
  
pkg load database
conn = pq_connect(setdbopts('dbname','postgres','host','localhost','port','5432','user','postgres','password','2807289050109'))

disp("1. Cálculo de iva")
disp("2. Historial")
main=input("Opción: ");
switch main
 case 1
 precio=input("Precio: ");
 noiva = (precio/1.12) ;
 iva = precio - noiva;
 disp("Precio sin iva: ")
 disp(noiva)
 disp("Precio del iva: ")
 disp(iva)
 pq_exec_params(conn, 'insert into programa3 values ($1,$2,$3);',{precio,noiva,iva});
 case 2
 pq_exec_params (conn, "select * from programa3;")
 otherwise
 disp("Escoge una opción válida")
end
catch
 disp("Algo salió mal")
end_try_catch

File: P3.py
import psycopg2

try: 
    conexion = psycopg2.connect(
        host = "localhost",
        port = "5432",
        user = "postgres",
        password = "2807289050109",
        dbname = "postgres"
    )
    print("Conexión exitosa")
except psycopg2.Error as e:
    print("Ocurrió un error en la conexión")
    print("Verifique los parámetros")

def main():

    print ("\nMenú ")
    print ("\n1.Calcular iva")
    print ("2.Historial")

    opcion = input("\nEscoge la opción deseada: ")

    if opcion == "1":

        def calculoiva():
            precio = float(input("Precio: "))
            cursor = conexion.cursor()
            noiva = (precio/1.12)
            iva = precio-noiva
            print("Precio sin iva: ", noiva)
            print("Precio del iva: ",iva)
            cursor.execute("insert into programa3(precio,sin_iva,iva) values(%s, %s,%s);",(precio,noiva,iva))
            conexion.commit()
            cursor.close()
            conexion.close()
        calculoiva()

    if opcion == "2":

        cursor = conexion.cursor()
        SQL = 'select * from programa3;'
        cursor.execute(SQL)
        valores = cursor.fetchall()
        print(valores)
main()